libtenzir/src/importer.cpp

```diff
@@ -134,60 +133,16 @@ importer_state::status(status_verbosity v) const {
   return rs->promise;
 }
 
-void importer_state::send_report() {
-  auto now = stopwatch::now();
-  using namespace std::string_literals;
-  auto elapsed = std::chrono::duration_cast<duration>(now - last_report);
-  auto node_throughput = measurement{elapsed, measurement_.events};
-  auto num_schemas_seen = schema_counters.size();
-  std::vector<performance_sample> samples = {};
-  samples.reserve(num_schemas_seen);
-  samples.push_back(performance_sample{"importer"s, measurement_});
-  samples.push_back(performance_sample{"node_throughput"s, node_throughput});
-  auto total_count = uint64_t{0};
-  for (const auto& [name, count] : schema_counters) {
-    total_count += count;
-    samples.push_back(performance_sample{
-      "ingest", measurement{elapsed, count}, {{"schema", name}}});
-  }
-  auto total_measurement = measurement{elapsed, total_count};
-  samples.push_back(performance_sample{"ingest-total"s, total_measurement});
-  schema_counters.clear();
-  auto r = performance_report{
-    .data = std::move(samples),
-  };
-#if TENZIR_LOG_LEVEL >= TENZIR_LOG_LEVEL_VERBOSE
-  auto beat = [&](const auto& sample) {
-    if (sample.value.events > 0) {
-      if (auto rate = sample.value.rate_per_sec(); std::isfinite(rate)) {
-        TENZIR_VERBOSE("{} handled {} events at a rate of {} events/sec in {}",
-                       *self, sample.value.events, static_cast<uint64_t>(rate),
-                       to_string(sample.value.duration));
-      } else {
-        TENZIR_VERBOSE("{} handled {} events in {}", *self, sample.value.events,
-                       to_string(sample.value.duration));
-      }
-    }
-  };
-  beat(r.data[1]);
-#endif
-  measurement_ = measurement{};
-  self->send(accountant, atom::metrics_v, std::move(r));
-  last_report = now;
-}
-
 void importer_state::on_process(const table_slice& slice) {
-  TENZIR_ASSERT(slice.rows() > 0);
+  const auto rows = slice.rows();
+  TENZIR_ASSERT(rows > 0);
   auto t = timer::start(measurement_);
-  auto rows = slice.rows();
-  auto name = slice.schema().name();
-  if (auto it = schema_counters.find(name); it != schema_counters.end()) {
-    it.value() += rows;
-  } else {
-    schema_counters.emplace(std::string{name}, rows);
+  const auto is_internal = slice.schema().attribute("internal").has_value();
+  if (not is_internal) {
+    schema_counters[slice.schema()] += rows;
   }
```

Comment:
Interesting that you exclude internal events. What's the rationale behind that?

Reply:
We do this for all metrics that work with events to avoid cycles. We can always re-evaluate this later, but for now this is just consistent with the other metrics.